test/internal/memcached_connection.py

```diff
@@ -238,7 +239,10 @@ def _read_and_parse_response(self):
             if is_indecrq(op):
                 val = INDECR_STRUCT.unpack_from(val)[0]
         if val is not None:
-            retval['val'] = val
+            if isinstance(val, bytes):
+                retval['val'] = bytes_to_str(val)
+            else:
+                retval['val'] = val
```

Comment:
Feel free to resolve this comment without changes too, but I'll tell what I think here.

[python-memcached](https://github.com/linsomniac/python-memcached) and [python-binary-memcached](https://github.com/jaysonsantos/python-binary-memcached) set a test/binary flag on writting a value and lean on the flag to decode into bytes or text at reading. It is quite intuitive for Python's client. However it is not wortful to implement for our testing implementation.

So here we agree on unconditional interpreting values as text ones. It would be good to mention this fact somewhere.

Next, why I even looked at this hunk of the code? Here we don't use our understanding of existing code to make the new code simpler. What I can deduce from your variant of the code? `val` is either bytes or something else. But what else? A string? Hm... I need to read `_read_response()` and `_recv()` to understand what is going on here.

If we'll rewrite it in a more strict way:

```python
if val_lenv is None or val_len > 0:
    begin = HEADER_SIZE + ext_len + key_len
    end   = HEADER_SIZE + tot_len
    val   = cmd[begin:end] 
    if is_indecrq(op): 
        # incr[q]/decr[q] result is integer.
        val = INDECR_STRUCT.unpack_from(val)[0]
    else:
        # Ideally we should lean on 4th bit in flags to
        # decide whether to interpret this value as binary
        # or as text.
        # 
        # Unconditional interpreting it as a text is
        # enough for testing purposes, though.
        val = bytes_to_str(val)
    retval['val'] = val
```

Everything becomes simple. If there is a value in the answer, we decode it as integer for increment / decrement and decode it as a text otherwise. No need to bury into other code.

It is stylistic, of course, but interesting. Ever if you'll not change it, please, share, what do you think?

Reply:
Updated, thanks!

```diff
diff --git a/test/internal/memcached_connection.py b/test/internal/memcached_connection.py
index 2b742fa..a707f1b 100644
--- a/test/internal/memcached_connection.py
+++ b/test/internal/memcached_connection.py
@@ -238,11 +238,15 @@ class MemcachedBinaryConnection(TarantoolConnection):
             # decode result of (incr/decr)(q)
             if is_indecrq(op):
                 val = INDECR_STRUCT.unpack_from(val)[0]
-        if val is not None:
-            if isinstance(val, bytes):
-                retval['val'] = bytes_to_str(val)
             else:
-                retval['val'] = val
+                # Ideally we should lean on 4th bit in flags to
+                # decide whether to interpret this value as binary
+                # or as text.
+                #
+                # Unconditional interpreting it as a text is
+                # enough for testing purposes, though.
+                val = bytes_to_str(val)
+            retval['val'] = val
 
         return retval
 
```